### DISCUSSION TITLE
issue in finding html views

### DISCUSSION THREAD

**Discussion by abdultkd** [processone/ejabberd#3751] [Questions] [answered]
I am beginner in erlang and want to learn from this project . I have searched in all directories of this project but found only 2 html files one for start and one for end server. can anyone help me to locate other html pages or hrl files

**Comment by licaon-kter** [comment]
Try https://docs.ejabberd.im ?

**Reply by abdultkd** [reply]
but where i can find views that are created for this application

**Comment by licaon-kter** [comment]
This sounds like you're confusing some stuff.

There are some web pages but only in the Admin area, eg: https://github.com/processone/ejabberd/tree/master/priv

What do you want to see exactly? This is an XMPP server, it needs an actual client that connects. If you want an HTML client see https://conversejs.org/ or https://movim.eu/

**Reply by abdultkd** [reply]
let me elaborate my self I am interested in admin panel views

**Comment by weiss** [comment]
> I am beginner in erlang and want to learn from this project

ejabberd is quite a large and complex project with a long history and support for some legacy build systems, etc.  If you're looking for an easy example to learn from, I'd probably suggest something simpler to begin with (e.g., [eturnal][1], though that doesn't have any web stuff at all).  For playing around with a project, I find the [rebar3 shell][2] very helpful, by the way.

Anyway, as for your actual question, it's unclear what "html pages or hrl files" you're talking about, as @licaon-kter said.

[1]: https://eturnal.net
[2]: https://ferd.ca/rebar3-shell.html

**Reply by abdultkd** [reply]
Thankyou weiss but as I have to work on web sip server thats why it is an ideal project for me i think

**Comment by abdultkd** [comment]
kindly if someone guide me about admin panel views like user management and other modules

**Comment by weiss** [comment, marked as answer]
> admin panel views like user management and other modules

ejabberd ships [ejabberd_web_admin][1] which servers a (somewhat minimalistic) web interface directly from the ejabberd node (using [ejabberd_http][2]).  Apart from that, ejabberd offers a [REST API][3] that can be used to build web interfaces outside Erlang.

[1]: https://github.com/processone/ejabberd/blob/master/src/ejabberd_web_admin.erl
[2]: https://github.com/processone/ejabberd/blob/master/src/ejabberd_http.erl
[3]: https://docs.ejabberd.im/developer/ejabberd-api/

**Comment by licaon-kter** [comment]
> web sip server

Sounds confusing, so it's a "web server" or a "sip server" or a "web interface for calls (and calls are via SIP)"?
